From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Damien Caliste <dcaliste@free.fr>
Date: Wed, 19 May 2021 08:59:40 +0200
Subject: [PATCH] Adjust for Qt < 5.14

Revert "Use range constructors for lists and sets"
This reverts commit 260cb748a7556b1e6a70efa05522c01200dda823.

Revert "Replace deprecated QString::SplitBehavior"
This reverts commit faf30f7544c7a1f5ec07ac68d1f0b4fbf4ed855d.

Revert "Fix disappearance of QDateTime(QDate)"
This reverts commit a8db671f00ebf5aa56e57d1bdf1511d9af22c809.
---
 src/libraries/qmfclient/qmailmessagekey.cpp   |  9 +--
 .../qmfclient/qmailmessagelistmodel.cpp       |  4 +-
 src/libraries/qmfclient/qmailmessageset.cpp   | 15 ++---
 .../qmfclient/qmailmessagethreadedmodel.cpp   | 14 ++---
 src/libraries/qmfclient/qmailstore.cpp        | 15 ++---
 .../qmfclient/qmailstorenotifier_p.cpp        | 18 +++---
 src/libraries/qmfclient/qmailtimestamp.cpp    |  2 +-
 .../qmfclient/support/qmailnamespace.cpp      | 12 ++--
 .../imap/imapconfiguration.cpp                |  2 +-
 .../messageservices/imap/imapprotocol.cpp     | 10 ++--
 .../messageservices/imap/imapservice.cpp      |  2 +-
 .../messageservices/imap/imapstrategy.cpp     | 24 ++++----
 .../messageservices/imap/integerregion.cpp    |  2 +-
 .../messageservices/smtp/smtpclient.cpp       |  2 +-
 src/tools/messageserver/servicehandler.cpp    | 39 +++++-------
 src/tools/messageserver/servicehandler.h      |  1 -
 .../tst_qmail_listmodels.cpp                  |  8 +--
 .../tst_qmail_sortkeys/tst_qmail_sortkeys.cpp | 20 +++----
 .../tst_qmaildisconnected.cpp                 | 22 +++----
 .../tst_qmailmessageset.cpp                   | 20 +++----
 .../tst_qmailserviceaction.cpp                | 22 +++----
 .../tst_qmailstorageaction.cpp                | 32 +++++-----
 tests/tst_qmailstore/tst_qmailstore.cpp       |  4 +-
 .../tst_qmailstorekeys/tst_qmailstorekeys.cpp | 60 +++++++++----------
 tests/tst_qmailthread/tst_qmailthread.cpp     | 20 +++----
 .../tst_storagemanager/tst_storagemanager.cpp | 16 ++---
 26 files changed, 179 insertions(+), 216 deletions(-)

diff --git a/src/libraries/qmfclient/qmailmessagekey.cpp b/src/libraries/qmfclient/qmailmessagekey.cpp
index 31f8541c..7be74e21 100644
--- a/src/libraries/qmfclient/qmailmessagekey.cpp
+++ b/src/libraries/qmfclient/qmailmessagekey.cpp
@@ -378,8 +378,7 @@ QMailMessageKey QMailMessageKey::id(const QMailMessageIdList &ids, QMailDataComp
     if (ids.count() >= IdLookupThreshold) {
         // If there are a large number of IDs, they will be inserted into a temporary table
         // with a uniqueness constraint; ensure only unique values are supplied
-        const QSet<QMailMessageId> uniqueIds(ids.constBegin(), ids.constEnd());
-        return QMailMessageKey(uniqueIds.values(), Id, QMailKey::comparator(cmp));
+        return QMailMessageKey(ids.toSet().toList(), Id, QMailKey::comparator(cmp));
     }
 
     return QMailMessageKey(ids, Id, QMailKey::comparator(cmp));
@@ -657,8 +656,7 @@ QMailMessageKey QMailMessageKey::serverUid(const QStringList &uids, QMailDataCom
     if (uids.count() >= IdLookupThreshold) {
         // If there are a large number of UIDs, they will be inserted into a temporary table
         // with a uniqueness constraint; ensure only unique values are supplied
-        const QSet<QString> uniqueIds(uids.constBegin(), uids.constEnd());
-        return QMailMessageKey(uniqueIds.values(), ServerUid, QMailKey::comparator(cmp));
+        return QMailMessageKey(uids.toSet().toList(), ServerUid, QMailKey::comparator(cmp));
     }
 
     return QMailMessageKey(uids, ServerUid, QMailKey::comparator(cmp));
@@ -989,8 +987,7 @@ QMailMessageKey QMailMessageKey::copyServerUid(const QStringList &uids, QMailDat
     if (uids.count() >= IdLookupThreshold) {
         // If there are a large number of UIDs, they will be inserted into a temporary table
         // with a uniqueness constraint; ensure only unique values are supplied
-        const QSet<QString> uniqueIds(uids.constBegin(), uids.constEnd());
-        return QMailMessageKey(uniqueIds.values(), CopyServerUid, QMailKey::comparator(cmp));
+        return QMailMessageKey(uids.toSet().toList(), CopyServerUid, QMailKey::comparator(cmp));
     }
 
     return QMailMessageKey(uids, CopyServerUid, QMailKey::comparator(cmp));
diff --git a/src/libraries/qmfclient/qmailmessagelistmodel.cpp b/src/libraries/qmfclient/qmailmessagelistmodel.cpp
index d8f69875..2b1dd8b1 100644
--- a/src/libraries/qmfclient/qmailmessagelistmodel.cpp
+++ b/src/libraries/qmfclient/qmailmessagelistmodel.cpp
@@ -402,9 +402,7 @@ bool QMailMessageListModelPrivate::updateMessages(const QMailMessageIdList &ids)
     QList<int> updateIndices;
 
     // Find the updated positions for our messages
-    QSet<QMailMessageId> uniqueIds(_idList.constBegin(), _idList.constEnd());
-    uniqueIds.unite(QSet<QMailMessageId>(ids.constBegin(), ids.constEnd()));
-    QMailMessageKey idKey(QMailMessageKey::id(uniqueIds.values()));
+    QMailMessageKey idKey(QMailMessageKey::id((_idList.toSet() + ids.toSet()).toList()));
     QMailMessageIdList newIds(QMailStore::instance()->queryMessages(_key & idKey, _sortKey, _limit));
     QMap<QMailMessageId, int> newPositions;
 
diff --git a/src/libraries/qmfclient/qmailmessageset.cpp b/src/libraries/qmfclient/qmailmessageset.cpp
index 72242950..12b57a4d 100644
--- a/src/libraries/qmfclient/qmailmessageset.cpp
+++ b/src/libraries/qmfclient/qmailmessageset.cpp
@@ -976,7 +976,7 @@ void QMailFilterMessageSet::messagesAdded(const QMailMessageIdList &ids)
         QMailMessageIdList matchingIds = QMailStore::instance()->queryMessages(key & idFilter);
         if (!matchingIds.isEmpty()) {
             // Our filtered message set has changed
-            d->_messageIds.unite(QSet<QMailMessageId>(matchingIds.constBegin(), matchingIds.constEnd()));
+            d->_messageIds.unite(QSet<QMailMessageId>::fromList(matchingIds));
             update(this);
         }
     }
@@ -989,7 +989,7 @@ void QMailFilterMessageSet::messagesRemoved(const QMailMessageIdList &ids)
 
     QSet<QMailMessageId>& _messageIds = d->_messageIds;
     if (!_messageIds.isEmpty()) {
-        QSet<QMailMessageId> removedIds = QSet<QMailMessageId>(ids.constBegin(), ids.constEnd());
+        QSet<QMailMessageId> removedIds = QSet<QMailMessageId>::fromList(ids);
 
         // See if any of these messages are in our set
         removedIds.intersect(_messageIds);
@@ -1008,12 +1008,11 @@ void QMailFilterMessageSet::messagesUpdated(const QMailMessageIdList &ids)
     QMailMessageKey key(messageKey());
     if (!key.isNonMatching()) {
         QSet<QMailMessageId>& _messageIds = d->_messageIds;
-        QSet<QMailMessageId> updatedIds = QSet<QMailMessageId>(ids.constBegin(), ids.constEnd());
+        QSet<QMailMessageId> updatedIds = QSet<QMailMessageId>::fromList(ids);
 
         // Find which of the updated messages should be in our set
         QMailMessageKey idFilter(QMailMessageKey::id(ids));
-        const QMailMessageIdList filteredIds = QMailStore::instance()->queryMessages(key & idFilter);
-        QSet<QMailMessageId> matchingIds = QSet<QMailMessageId>(filteredIds.constBegin(), filteredIds.constEnd());
+        QSet<QMailMessageId> matchingIds = QSet<QMailMessageId>::fromList(QMailStore::instance()->queryMessages(key & idFilter));
 
         QSet<QMailMessageId> presentIds = updatedIds;
         QSet<QMailMessageId> absentIds = updatedIds;
@@ -1062,8 +1061,7 @@ void QMailFilterMessageSet::resyncState()
     Q_D(QMailFilterMessageSet);
 
     if (d->_minimized) {
-        const QMailMessageIdList ids = QMailStore::instance()->queryMessages(messageKey());
-        d->_messageIds = QSet<QMailMessageId>(ids.constBegin(), ids.constEnd());
+        d->_messageIds = QSet<QMailMessageId>::fromList(QMailStore::instance()->queryMessages(messageKey()));
     } else {
         d->_messageIds.clear();
     }
@@ -1085,8 +1083,7 @@ void QMailFilterMessageSet::reset()
     if (d->_minimized) {
         disconnect(model(), SIGNAL(folderContentsModified(QMailFolderIdList)), this, SLOT(folderContentsModified(QMailFolderIdList)));
 
-        const QMailMessageIdList ids = QMailStore::instance()->queryMessages(messageKey());
-        d->_messageIds = QSet<QMailMessageId>(ids.constBegin(), ids.constEnd());
+        d->_messageIds = QSet<QMailMessageId>::fromList(QMailStore::instance()->queryMessages(messageKey()));
 
         connect(model(), SIGNAL(messagesAdded(QMailMessageIdList)), this, SLOT(messagesAdded(QMailMessageIdList)));
         connect(model(), SIGNAL(messagesRemoved(QMailMessageIdList)), this, SLOT(messagesRemoved(QMailMessageIdList)));
diff --git a/src/libraries/qmfclient/qmailmessagethreadedmodel.cpp b/src/libraries/qmfclient/qmailmessagethreadedmodel.cpp
index 452f7c55..4eb5afa3 100644
--- a/src/libraries/qmfclient/qmailmessagethreadedmodel.cpp
+++ b/src/libraries/qmfclient/qmailmessagethreadedmodel.cpp
@@ -543,13 +543,12 @@ bool QMailMessageThreadedModelPrivate::processMessagesUpdated(const QMailMessage
 
 bool QMailMessageThreadedModelPrivate::updateMessages(const QMailMessageIdList &ids)
 {
-    QSet<QMailMessageId> existingIds(_currentIds.constBegin(), _currentIds.constEnd());
-    existingIds.unite(QSet<QMailMessageId>(ids.constBegin(), ids.constEnd()));
+    QSet<QMailMessageId> existingIds(_currentIds.toSet());
 
-    QMailMessageKey idKey(QMailMessageKey::id(existingIds.values()));
+    QMailMessageKey idKey(QMailMessageKey::id((existingIds + ids.toSet()).toList()));
     QMailMessageIdList newIds(QMailStore::instance()->queryMessages(_key & idKey, _sortKey, _limit));
 
-    QSet<QMailMessageId> currentIds(newIds.constBegin(), newIds.constEnd());
+    QSet<QMailMessageId> currentIds(newIds.toSet());
 
     // Find which of the messages we must add and remove
     QMailMessageIdList additionIds;
@@ -649,10 +648,7 @@ bool QMailMessageThreadedModelPrivate::updateMessages(const QMailMessageIdList &
     removeMessages(temporaryRemovalIds, &readditionIds);
 
     // Find the locations for the added and reinserted messages
-    QSet<QMailMessageId> uniqueIds(additionIds.constBegin(), additionIds.constEnd());
-    uniqueIds.unite(QSet<QMailMessageId>(temporaryRemovalIds.constBegin(), temporaryRemovalIds.constEnd()));
-    uniqueIds.unite(QSet<QMailMessageId>(readditionIds.constBegin(), readditionIds.constEnd()));
-    addMessages(uniqueIds.values());
+    addMessages((additionIds.toSet() + temporaryRemovalIds.toSet() + readditionIds.toSet()).toList());
 
     return true;
 }
@@ -724,7 +720,7 @@ bool QMailMessageThreadedModelPrivate::removeMessages(const QMailMessageIdList &
     }
 
     if (readditions) {
-        *readditions = childIds.values();
+        *readditions = childIds.toList();
     }
 
     return true;
diff --git a/src/libraries/qmfclient/qmailstore.cpp b/src/libraries/qmfclient/qmailstore.cpp
index b85f4a1e..9f1b8135 100644
--- a/src/libraries/qmfclient/qmailstore.cpp
+++ b/src/libraries/qmfclient/qmailstore.cpp
@@ -1255,8 +1255,7 @@ void QMailStore::emitAccountNotification(ChangeType type, const QMailAccountIdLi
     Q_ASSERT(!ids.contains(QMailAccountId()));
     if (!ids.isEmpty()) {
         // Ensure there are no duplicates in the list
-        const QSet<QMailAccountId> uids(ids.constBegin(), ids.constEnd());
-        QMailAccountIdList idList(uids.constBegin(), uids.constEnd());
+        QMailAccountIdList idList(ids.toSet().toList());
 
         d->notifyAccountsChange(type, idList);
 
@@ -1286,8 +1285,7 @@ void QMailStore::emitFolderNotification(ChangeType type, const QMailFolderIdList
     Q_ASSERT(!ids.contains(QMailFolderId()));
     if (!ids.isEmpty()) {
         // Ensure there are no duplicates in the list
-        const QSet<QMailFolderId> uids(ids.constBegin(), ids.constEnd());
-        QMailFolderIdList idList(uids.constBegin(), uids.constEnd());
+        QMailFolderIdList idList(ids.toSet().toList());
 
         d->notifyFoldersChange(type, idList);
 
@@ -1317,8 +1315,7 @@ void QMailStore::emitThreadNotification(ChangeType type, const QMailThreadIdList
     Q_ASSERT(!ids.contains(QMailThreadId()));
     if (!ids.isEmpty()) {
         // Ensure there are no duplicates in the list
-        const QSet<QMailThreadId> uids(ids.constBegin(), ids.constEnd());
-        QMailThreadIdList idList(uids.constBegin(), uids.constEnd());
+        QMailThreadIdList idList(ids.toSet().toList());
 
         d->notifyThreadsChange(type, idList);
 
@@ -1348,8 +1345,7 @@ void QMailStore::emitMessageNotification(ChangeType type, const QMailMessageIdLi
     Q_ASSERT(!ids.contains(QMailMessageId()));
     if (!ids.isEmpty()) {
         // Ensure there are no duplicates in the list
-        const QSet<QMailMessageId> uids(ids.constBegin(), ids.constEnd());
-        QMailMessageIdList idList(uids.constBegin(), uids.constEnd());
+        QMailMessageIdList idList(ids.toSet().toList());
 
         d->notifyMessagesChange(type, idList);
 
@@ -1420,8 +1416,7 @@ void QMailStore::emitRemovalRecordNotification(ChangeType type, const QMailAccou
 {
     if (!ids.isEmpty()) {
         // Ensure there are no duplicates in the list
-        const QSet<QMailAccountId> uids(ids.constBegin(), ids.constEnd());
-        QMailAccountIdList idList(uids.constBegin(), uids.constEnd());
+        QMailAccountIdList idList(ids.toSet().toList());
 
         d->notifyMessageRemovalRecordsChange(type, idList);
 
diff --git a/src/libraries/qmfclient/qmailstorenotifier_p.cpp b/src/libraries/qmfclient/qmailstorenotifier_p.cpp
index 8a2fbe13..87508cec 100644
--- a/src/libraries/qmfclient/qmailstorenotifier_p.cpp
+++ b/src/libraries/qmfclient/qmailstorenotifier_p.cpp
@@ -174,7 +174,7 @@ void QMailStoreNotifier::notifyAccountsChange(QMailStore::ChangeType changeType,
             flushTimer.start();
         }
 
-        QSet<QMailAccountId> idsSet = QSet<QMailAccountId>(ids.constBegin(), ids.constEnd());
+        QSet<QMailAccountId> idsSet = QSet<QMailAccountId>::fromList(ids);
         switch (changeType)
         {
         case QMailStore::Added:
@@ -209,7 +209,7 @@ void QMailStoreNotifier::notifyMessagesChange(QMailStore::ChangeType changeType,
             flushTimer.start();
         }
 
-        QSet<QMailMessageId> idsSet = QSet<QMailMessageId>(ids.constBegin(), ids.constEnd());
+        QSet<QMailMessageId> idsSet = QSet<QMailMessageId>::fromList(ids);
         switch (changeType)
         {
         case QMailStore::Added:
@@ -283,7 +283,7 @@ void QMailStoreNotifier::notifyMessagesDataChange(const QMailMessageIdList& ids,
             flushTimer.start();
         }
 
-        MessagesProperties props(QPair<QMailMessageKey::Properties, QMailMessageMetaData>(properties, data), QSet<QMailMessageId>(ids.constBegin(), ids.constEnd()));
+        MessagesProperties props(QPair<QMailMessageKey::Properties, QMailMessageMetaData>(properties, data), ids.toSet());
         messagesPropertiesBuffer.append(props);
 
     } else {
@@ -303,7 +303,7 @@ void QMailStoreNotifier::notifyMessagesDataChange(const QMailMessageIdList& ids,
         }
 
         MessagesStatus messageStatus(status, set);
-        messagesStatusBuffer[messageStatus] += QSet<QMailMessageId>(ids.constBegin(), ids.constEnd());
+        messagesStatusBuffer[messageStatus] += ids.toSet();
 
     } else {
         emit ipcAdaptor->messageStatusChanged(ids, status, set);
@@ -321,7 +321,7 @@ void QMailStoreNotifier::notifyThreadsChange(QMailStore::ChangeType changeType,
             flushTimer.start();
         }
 
-        QSet<QMailThreadId> idsSet = QSet<QMailThreadId>(ids.constBegin(), ids.constEnd());
+        QSet<QMailThreadId> idsSet = QSet<QMailThreadId>::fromList(ids);
         switch (changeType)
         {
         case QMailStore::Added:
@@ -357,7 +357,7 @@ void QMailStoreNotifier::notifyFoldersChange(QMailStore::ChangeType changeType,
             flushTimer.start();
         }
 
-        QSet<QMailFolderId> idsSet = QSet<QMailFolderId>(ids.constBegin(), ids.constEnd());
+        QSet<QMailFolderId> idsSet = QSet<QMailFolderId>::fromList(ids);
         switch (changeType)
         {
         case QMailStore::Added:
@@ -392,7 +392,7 @@ void QMailStoreNotifier::notifyMessageRemovalRecordsChange(QMailStore::ChangeTyp
             flushTimer.start();
         }
 
-        QSet<QMailAccountId> idsSet = QSet<QMailAccountId>(ids.constBegin(), ids.constEnd());
+        QSet<QMailAccountId> idsSet = QSet<QMailAccountId>::fromList(ids);
         switch (changeType)
         {
         case QMailStore::Added:
@@ -430,7 +430,7 @@ void QMailStoreNotifier::notifyTransmissionInProgress(const QMailAccountIdList&
 
 bool QMailStoreNotifier::setRetrievalInProgress(const QMailAccountIdList& ids)
 {
-    QSet<QMailAccountId> idSet(ids.constBegin(), ids.constEnd());
+    QSet<QMailAccountId> idSet(ids.toSet());
     if ((idSet != retrievalInProgressIds) || !retrievalSetInitialized) {
         retrievalInProgressIds = idSet;
         retrievalSetInitialized = true;
@@ -442,7 +442,7 @@ bool QMailStoreNotifier::setRetrievalInProgress(const QMailAccountIdList& ids)
 
 bool QMailStoreNotifier::setTransmissionInProgress(const QMailAccountIdList& ids)
 {
-    QSet<QMailAccountId> idSet(ids.constBegin(), ids.constEnd());
+    QSet<QMailAccountId> idSet(ids.toSet());
     if ((idSet != transmissionInProgressIds) || !transmissionSetInitialized) {
         transmissionInProgressIds = idSet;
         transmissionSetInitialized = true;
diff --git a/src/libraries/qmfclient/qmailtimestamp.cpp b/src/libraries/qmfclient/qmailtimestamp.cpp
index bcb18710..3f4a60da 100644
--- a/src/libraries/qmfclient/qmailtimestamp.cpp
+++ b/src/libraries/qmfclient/qmailtimestamp.cpp
@@ -120,7 +120,7 @@ QMailTimeStampPrivate::QMailTimeStampPrivate(const QString& timeText)
 
     // Extract the date/time elements
     uncommented = uncommented.trimmed();
-    QStringList tokens = uncommented.split(QChar::Space, Qt::SkipEmptyParts);
+    QStringList tokens = uncommented.split(QChar::Space, QString::SkipEmptyParts);
 
     int tokenCount = tokens.count();
     if ( tokenCount > 0 ) {
diff --git a/src/libraries/qmfclient/support/qmailnamespace.cpp b/src/libraries/qmfclient/support/qmailnamespace.cpp
index 45240aa0..2e30986d 100644
--- a/src/libraries/qmfclient/support/qmailnamespace.cpp
+++ b/src/libraries/qmfclient/support/qmailnamespace.cpp
@@ -388,24 +388,24 @@ static QMap<QByteArray, QStringList> standardFolderTranslations()
     QTextStream in(&file);
     while (!in.atEnd()) {
         QString line = in.readLine();
-        QStringList list = line.split(QLatin1Char('='), Qt::SkipEmptyParts);
+        QStringList list = line.split(QLatin1Char('='), QString::SkipEmptyParts);
         QString folderName = list.at(0);
         QString transList = list.at(1);
 
         if (folderName == QLatin1String("inbox")) {
-            QStringList inboxList = transList.split(QLatin1Char(','), Qt::SkipEmptyParts);
+            QStringList inboxList = transList.split(QLatin1Char(','), QString::SkipEmptyParts);
             folderTranslations.insert("inbox", inboxList);
         } else if (folderName == QLatin1String("drafts")) {
-            QStringList draftsList = transList.split(QLatin1Char(','), Qt::SkipEmptyParts);
+            QStringList draftsList = transList.split(QLatin1Char(','), QString::SkipEmptyParts);
             folderTranslations.insert("drafts", draftsList);
         } else if (folderName == QLatin1String("trash")) {
-            QStringList trashList = transList.split(QLatin1Char(','), Qt::SkipEmptyParts);
+            QStringList trashList = transList.split(QLatin1Char(','), QString::SkipEmptyParts);
             folderTranslations.insert("trash", trashList);
         } else if (folderName == QLatin1String("sent")) {
-            QStringList sentList = transList.split(QLatin1Char(','), Qt::SkipEmptyParts);
+            QStringList sentList = transList.split(QLatin1Char(','), QString::SkipEmptyParts);
             folderTranslations.insert("sent", sentList);
         } else if (folderName == QLatin1String("spam")) {
-            QStringList spamList = transList.split(QLatin1Char(','), Qt::SkipEmptyParts);
+            QStringList spamList = transList.split(QLatin1Char(','), QString::SkipEmptyParts);
             folderTranslations.insert("spam", spamList);
         }
     }
diff --git a/src/plugins/messageservices/imap/imapconfiguration.cpp b/src/plugins/messageservices/imap/imapconfiguration.cpp
index 810bfe75..99b4fbeb 100644
--- a/src/plugins/messageservices/imap/imapconfiguration.cpp
+++ b/src/plugins/messageservices/imap/imapconfiguration.cpp
@@ -145,7 +145,7 @@ bool ImapConfiguration::intervalCheckRoamingEnabled() const
 
 QStringList ImapConfiguration::capabilities() const
 {
-    return value("capabilities").split(QChar(' '), Qt::SkipEmptyParts);
+    return value("capabilities").split(QChar(' '), QString::SkipEmptyParts);
 }
 
 void ImapConfiguration::setCapabilities(const QStringList &s)
diff --git a/src/plugins/messageservices/imap/imapprotocol.cpp b/src/plugins/messageservices/imap/imapprotocol.cpp
index c16339e1..21a612c1 100644
--- a/src/plugins/messageservices/imap/imapprotocol.cpp
+++ b/src/plugins/messageservices/imap/imapprotocol.cpp
@@ -533,10 +533,10 @@ void ImapState::untaggedResponse(ImapContext *c, const QString &line)
     } else if (line.indexOf("[CAPABILITY", 0) != -1) {
         int start = 0;
         QString temp = token(line, '[', ']', &start);
-        QStringList capabilities = temp.mid(12).trimmed().split(' ', Qt::SkipEmptyParts);
+        QStringList capabilities = temp.mid(12).trimmed().split(' ', QString::SkipEmptyParts);
         c->protocol()->setCapabilities(capabilities);
     } else if (line.indexOf("* CAPABILITY ", 0) != -1) {
-        QStringList capabilities = line.mid(13).trimmed().split(' ', Qt::SkipEmptyParts);
+        QStringList capabilities = line.mid(13).trimmed().split(' ', QString::SkipEmptyParts);
         c->protocol()->setCapabilities(capabilities);
     }
 
@@ -656,7 +656,7 @@ void CapabilityState::untaggedResponse(ImapContext *c, const QString &line)
 {
     QStringList capabilities;
     if (line.startsWith(QLatin1String("* CAPABILITY"))) {
-        capabilities = line.mid(12).trimmed().split(' ', Qt::SkipEmptyParts);
+        capabilities = line.mid(12).trimmed().split(' ', QString::SkipEmptyParts);
         c->protocol()->setCapabilities(capabilities);
     } else {
         ImapState::untaggedResponse(c, line);
@@ -791,7 +791,7 @@ void LoginState::taggedResponse(ImapContext *c, const QString &line)
     if (line.indexOf("[CAPABILITY", Qt::CaseInsensitive) != -1) {
         int start = 0;
         QString temp = token(line, '[', ']', &start);
-        QStringList capabilities = temp.mid(12).trimmed().split(' ', Qt::SkipEmptyParts);
+        QStringList capabilities = temp.mid(12).trimmed().split(' ', QString::SkipEmptyParts);
         c->protocol()->setCapabilities(capabilities);
     }
 
@@ -1611,7 +1611,7 @@ void SelectedState::untaggedResponse(ImapContext *c, const QString &line)
     } else if (line.indexOf("PERMANENTFLAGS", 0, Qt::CaseInsensitive) != -1) {
         int start = 0;
         QString temp = token(line, '(', ')', &start);
-        c->setPermanentFlags(temp.split(' ', Qt::SkipEmptyParts));
+        c->setPermanentFlags(temp.split(' ', QString::SkipEmptyParts));
     } else if (line.indexOf("EXPUNGE", 0, Qt::CaseInsensitive) != -1) {
         quint32 exists = c->exists();
         if (exists > 0) {
diff --git a/src/plugins/messageservices/imap/imapservice.cpp b/src/plugins/messageservices/imap/imapservice.cpp
index 8d986a58..a4e8f0a8 100644
--- a/src/plugins/messageservices/imap/imapservice.cpp
+++ b/src/plugins/messageservices/imap/imapservice.cpp
@@ -1168,7 +1168,7 @@ bool ImapService::Source::prepareMessages(const QList<QPair<QMailMessagePart::Lo
         bool external(false);
 
         // Are these messages being resolved for internal or external references?
-        QMailMessageKey key(QMailMessageKey::id(referringIds.values()));
+        QMailMessageKey key(QMailMessageKey::id(referringIds.toList()));
         QMailMessageKey::Properties props(QMailMessageKey::Id | QMailMessageKey::ParentAccountId | QMailMessageKey::Status);
 
         for (const QMailMessageMetaData &metaData : QMailStore::instance()->messagesMetaData(key, props)) {
diff --git a/src/plugins/messageservices/imap/imapstrategy.cpp b/src/plugins/messageservices/imap/imapstrategy.cpp
index efb853eb..59863e7c 100644
--- a/src/plugins/messageservices/imap/imapstrategy.cpp
+++ b/src/plugins/messageservices/imap/imapstrategy.cpp
@@ -347,15 +347,13 @@ QSet<QMailFolderId> foldersApplicableTo(QMailMessageKey const& messagekey, QSet<
                     if (arg.op == QMailKey::Equal || arg.op == QMailKey::Includes) {
                         Q_ASSERT(arg.valueList.count() == 1);
                         Q_ASSERT(arg.valueList[0].canConvert<QMailFolderId>());
-                        const auto ancestorFolders = QMailStore::instance()->queryFolders(
-                                QMailFolderKey::ancestorFolderIds(arg.valueList[0].value<QMailFolderId>()));
-                        included.unite(QSet<QMailFolderId>(ancestorFolders.constBegin(), ancestorFolders.constEnd()));
+                        included.unite(QMailStore::instance()->queryFolders(
+                                QMailFolderKey::ancestorFolderIds(arg.valueList[0].value<QMailFolderId>())).toSet());
                     } else if (arg.op == QMailKey::NotEqual || arg.op == QMailKey::Excludes) {
                         Q_ASSERT(arg.valueList.count() == 1);
                         Q_ASSERT(arg.valueList[0].canConvert<QMailFolderId>());
-                        const auto ancestorFolders = QMailStore::instance()->queryFolders(
-                                QMailFolderKey::ancestorFolderIds(arg.valueList[0].value<QMailFolderId>()));
-                        excluded.unite(QSet<QMailFolderId>(ancestorFolders.constBegin(), ancestorFolders.constEnd()));
+                        excluded.unite(QMailStore::instance()->queryFolders(
+                                QMailFolderKey::ancestorFolderIds(arg.valueList[0].value<QMailFolderId>())).toSet());
                     } else {
                         Q_ASSERT(false);
                     }
@@ -365,15 +363,13 @@ QSet<QMailFolderId> foldersApplicableTo(QMailMessageKey const& messagekey, QSet<
                     if (arg.op == QMailKey::Equal || arg.op == QMailKey::Includes) {
                         Q_ASSERT(arg.valueList.count() == 1);
                         Q_ASSERT(arg.valueList[0].canConvert<QMailAccountId>());
-                        const auto parentAccountFolders = QMailStore::instance()->queryFolders(
-                                QMailFolderKey::parentAccountId(arg.valueList[0].value<QMailAccountId>()));
-                        included.unite(QSet<QMailFolderId>(parentAccountFolders.constBegin(), parentAccountFolders.constEnd()));
+                        included.unite(QMailStore::instance()->queryFolders(
+                                QMailFolderKey::parentAccountId(arg.valueList[0].value<QMailAccountId>())).toSet());
                     } else if (arg.op == QMailKey::NotEqual || arg.op == QMailKey::Excludes) {
                         Q_ASSERT(arg.valueList.count() == 1);
                         Q_ASSERT(arg.valueList[0].canConvert<QMailAccountId>());
-                        const auto parentAccountFolders = QMailStore::instance()->queryFolders(
-                                QMailFolderKey::parentAccountId(arg.valueList[0].value<QMailAccountId>()));
-                        excluded.unite(QSet<QMailFolderId>(parentAccountFolders.constBegin(), parentAccountFolders.constEnd()));
+                        excluded.unite(QMailStore::instance()->queryFolders(
+                                QMailFolderKey::parentAccountId(arg.valueList[0].value<QMailAccountId>())).toSet());
                     } else {
                         Q_ASSERT(false);
                     }
@@ -1908,9 +1904,9 @@ void ImapSearchMessageStrategy::folderListCompleted(ImapStrategyContextBase *con
         _limit = -1;
         _count = false;
     } else {
-        QSet<QMailFolderId> accountFolders(_mailboxList.constBegin(), _mailboxList.constEnd());
+        QSet<QMailFolderId> accountFolders(_mailboxList.toSet());
 
-        QMailFolderIdList foldersToSearch(foldersApplicableTo(_searches.first().criteria, accountFolders).values());
+        QMailFolderIdList foldersToSearch(foldersApplicableTo(_searches.first().criteria, accountFolders).toList());
 
         if (foldersToSearch.isEmpty()) {
             ImapRetrieveFolderListStrategy::folderListCompleted(context);
diff --git a/src/plugins/messageservices/imap/integerregion.cpp b/src/plugins/messageservices/imap/integerregion.cpp
index 3daa98a4..fde4d0e8 100644
--- a/src/plugins/messageservices/imap/integerregion.cpp
+++ b/src/plugins/messageservices/imap/integerregion.cpp
@@ -87,7 +87,7 @@ IntegerRegion::IntegerRegion(const QString &uidString)
 {
     // Performance note currently O(n^2), n = uids.count()
     // TODO: sort uids in uidString if they are not already sorted
-    QStringList rangeList = uidString.split(",", Qt::SkipEmptyParts);
+    QStringList rangeList = uidString.split(",", QString::SkipEmptyParts);
     foreach (const QString &s, rangeList) {
         bool ok = false;
         int index = s.indexOf(":");
diff --git a/src/plugins/messageservices/smtp/smtpclient.cpp b/src/plugins/messageservices/smtp/smtpclient.cpp
index e7ba6121..86bb828b 100644
--- a/src/plugins/messageservices/smtp/smtpclient.cpp
+++ b/src/plugins/messageservices/smtp/smtpclient.cpp
@@ -531,7 +531,7 @@ void SmtpClient::nextAction(const QString &response)
                     QStringList authCaps;
                     foreach (QString const& capability, capabilities) {
                         if (capability.startsWith("AUTH", Qt::CaseInsensitive)) {
-                            authCaps.append(capability.split(" ", Qt::SkipEmptyParts));
+                            authCaps.append(capability.split(" ", QString::SkipEmptyParts));
                         }
                     }
                     QMail::SaslMechanism authType = QMailAuthenticator::authFromCapabilities(authCaps);
diff --git a/src/tools/messageserver/servicehandler.cpp b/src/tools/messageserver/servicehandler.cpp
index da6f4f1a..3d633126 100644
--- a/src/tools/messageserver/servicehandler.cpp
+++ b/src/tools/messageserver/servicehandler.cpp
@@ -948,11 +948,6 @@ QSet<QMailMessageService*> ServiceHandler::sourceServiceSet(const QSet<QMailAcco
     return services;
 }
 
-QSet<QMailMessageService*> ServiceHandler::sourceServiceSet(const QMailAccountIdList &ids) const
-{
-    return sourceServiceSet(QSet<QMailAccountId>(ids.constBegin(), ids.constEnd()));
-}
-
 QSet<QMailMessageService*> ServiceHandler::sinkServiceSet(const QMailAccountId &id) const
 {
     QSet<QMailMessageService*> services;
@@ -1214,12 +1209,10 @@ void ServiceHandler::expireAction()
                     }
 
                     if (retrievalSetModified) {
-                        QMailStore::instance()->setRetrievalInProgress(QMailAccountIdList(_retrievalAccountIds.constBegin(),
-                                                                                          _retrievalAccountIds.constEnd()));
+                        QMailStore::instance()->setRetrievalInProgress(_retrievalAccountIds.toList());
                     }
                     if (transmissionSetModified) {
-                        QMailStore::instance()->setTransmissionInProgress(QMailAccountIdList(_transmissionAccountIds.constBegin(),
-                                                                                             _transmissionAccountIds.constEnd()));
+                        QMailStore::instance()->setTransmissionInProgress(_transmissionAccountIds.toList());
                     }
 
                     mActiveActions.erase(it);
@@ -1228,7 +1221,7 @@ void ServiceHandler::expireAction()
                 mActionExpiry.removeFirst();
 
                 // Restart the service(s) for each of these accounts
-                QMailAccountIdList ids(serviceAccounts.constBegin(), serviceAccounts.constEnd());
+                QMailAccountIdList ids(serviceAccounts.toList());
                 deregisterAccountServices(ids, QMailServiceAction::Status::ErrTimeout, tr("Request is not progressing"));
                 registerAccountServices(ids);
 
@@ -1293,12 +1286,10 @@ void ServiceHandler::cancelTransfer(quint64 action)
         }
 
         if (retrievalSetModified) {
-            QMailStore::instance()->setRetrievalInProgress(QMailAccountIdList(_retrievalAccountIds.constBegin(),
-                                                                              _retrievalAccountIds.constEnd()));
+            QMailStore::instance()->setRetrievalInProgress(_retrievalAccountIds.toList());
         }
         if (transmissionSetModified) {
-            QMailStore::instance()->setTransmissionInProgress(QMailAccountIdList(_transmissionAccountIds.constBegin(),
-                                                                                 _transmissionAccountIds.constEnd()));
+            QMailStore::instance()->setTransmissionInProgress(_transmissionAccountIds.toList());
         }
 
         //The ActionData might have already been deleted by actionCompleted, triggered by cancelOperation
@@ -1374,7 +1365,7 @@ void ServiceHandler::transmitMessages(quint64 action, const QMailAccountId &acco
                  QList<QPair<QMailMessagePart::Location,
                              QMailMessagePart::Location> > > unresolvedLists(messageResolvers(unresolvedMessages));
 
-            sources = sourceServiceSet(unresolvedLists.keys());
+            sources = sourceServiceSet(unresolvedLists.keys().toSet());
 
             // Emit no signal after completing preparation
             enqueueRequest(new PrepareMessagesRequest(unresolvedLists), action, sources, &ServiceHandler::dispatchPrepareMessages, 0);
@@ -1423,7 +1414,7 @@ void ServiceHandler::transmitMessage(quint64 action, const QMailMessageId &messa
                  QList<QPair<QMailMessagePart::Location,
                              QMailMessagePart::Location> > > unresolvedLists(messageResolvers(unresolvedMessages));
 
-            sources = sourceServiceSet(unresolvedLists.keys());
+            sources = sourceServiceSet(unresolvedLists.keys().toSet());
 
             // Emit no signal after completing preparation
             enqueueRequest(new PrepareMessagesRequest(unresolvedLists), action, sources, &ServiceHandler::dispatchPrepareMessages, 0);
@@ -1786,7 +1777,7 @@ void ServiceHandler::retrieveMessages(quint64 action, const QMailMessageIdList &
 {
     QMap<QMailAccountId, QMailMessageIdList> messageLists(accountMessages(messageIds));
 
-    QSet<QMailMessageService*> sources(sourceServiceSet(messageLists.keys()));
+    QSet<QMailMessageService*> sources(sourceServiceSet(messageLists.keys().toSet()));
     if (sources.isEmpty()) {
         reportFailure(action, QMailServiceAction::Status::ErrNoConnection,
                       tr("Unable to retrieve messages for unconfigured account"));
@@ -1821,7 +1812,7 @@ bool ServiceHandler::dispatchRetrieveMessages(Request *req)
         }
     }
 
-    QMailStore::instance()->setRetrievalInProgress(QMailAccountIdList(_retrievalAccountIds.constBegin(), _retrievalAccountIds.constEnd()));
+    QMailStore::instance()->setRetrievalInProgress(_retrievalAccountIds.toList());
     return true;
 }
 
@@ -2039,7 +2030,7 @@ void ServiceHandler::onlineDeleteMessages(quint64 action, const QMailMessageIdLi
         discardMessages(action, messageIds);
     } else {
         QMap<QMailAccountId, QMailMessageIdList> messageLists(accountMessages(messageIds));
-        sources = sourceServiceSet(messageLists.keys());
+        sources = sourceServiceSet(messageLists.keys().toSet());
         if (sources.isEmpty()) {
             reportFailure(action, QMailServiceAction::Status::ErrNoConnection, tr("Unable to delete messages for unconfigured account"));
         } else {
@@ -2230,7 +2221,7 @@ void ServiceHandler::onlineMoveMessages(quint64 action, const QMailMessageIdList
     QSet<QMailMessageService*> sources;
 
     QMap<QMailAccountId, QMailMessageIdList> messageLists(accountMessages(messageIds));
-    sources = sourceServiceSet(messageLists.keys());
+    sources = sourceServiceSet(messageLists.keys().toSet());
     if (sources.isEmpty()) {
         reportFailure(action, QMailServiceAction::Status::ErrNoConnection, tr("Unable to move messages for unconfigured account"));
     } else {
@@ -2283,7 +2274,7 @@ void ServiceHandler::onlineFlagMessagesAndMoveToStandardFolder(quint64 action, c
     QSet<QMailMessageService*> sources;
 
     QMap<QMailAccountId, QMailMessageIdList> messageLists(accountMessages(messageIds));
-    sources = sourceServiceSet(messageLists.keys());
+    sources = sourceServiceSet(messageLists.keys().toSet());
     if (sources.isEmpty()) {
         reportFailure(action, QMailServiceAction::Status::ErrNoConnection, tr("Unable to flag messages for unconfigured account"));
     } else {
@@ -2720,7 +2711,7 @@ void ServiceHandler::searchMessages(quint64 action, const QMailMessageKey &filte
 {
     if (spec == QMailSearchAction::Remote) {
         // Find the accounts that we need to search within from the criteria
-        QSet<QMailAccountId> searchAccountIds(accountsApplicableTo(filter, QSet<QMailAccountId>(sourceMap.keyBegin(), sourceMap.keyEnd())));
+        QSet<QMailAccountId> searchAccountIds(accountsApplicableTo(filter, sourceMap.keys().toSet()));
 
         QSet<QMailMessageService*> sources(sourceServiceSet(searchAccountIds));
         if (sources.isEmpty()) {
@@ -3310,7 +3301,7 @@ void ServiceHandler::setRetrievalInProgress(const QMailAccountId &accountId, boo
     }
 
     if (modified) {
-        QMailStore::instance()->setRetrievalInProgress(QMailAccountIdList(_retrievalAccountIds.constBegin(), _retrievalAccountIds.constEnd()));
+        QMailStore::instance()->setRetrievalInProgress(_retrievalAccountIds.toList());
     }
 }
 
@@ -3327,7 +3318,7 @@ void ServiceHandler::setTransmissionInProgress(const QMailAccountId &accountId,
     }
 
     if (modified) {
-        QMailStore::instance()->setTransmissionInProgress(QMailAccountIdList(_transmissionAccountIds.constBegin(), _transmissionAccountIds.constEnd()));
+        QMailStore::instance()->setTransmissionInProgress(_transmissionAccountIds.toList());
     }
 }
 
diff --git a/src/tools/messageserver/servicehandler.h b/src/tools/messageserver/servicehandler.h
index e873e801..f4230242 100644
--- a/src/tools/messageserver/servicehandler.h
+++ b/src/tools/messageserver/servicehandler.h
@@ -243,7 +243,6 @@ private:
 
     QSet<QMailMessageService*> sourceServiceSet(const QMailAccountId &id) const;
     QSet<QMailMessageService*> sourceServiceSet(const QSet<QMailAccountId> &ids) const;
-    QSet<QMailMessageService*> sourceServiceSet(const QMailAccountIdList &ids) const;
 
     QSet<QMailMessageService*> sinkServiceSet(const QMailAccountId &id) const;
     QSet<QMailMessageService*> sinkServiceSet(const QSet<QMailAccountId> &ids) const;
diff --git a/tests/tst_qmail_listmodels/tst_qmail_listmodels.cpp b/tests/tst_qmail_listmodels/tst_qmail_listmodels.cpp
index 6d6abd20..a908772d 100644
--- a/tests/tst_qmail_listmodels/tst_qmail_listmodels.cpp
+++ b/tests/tst_qmail_listmodels/tst_qmail_listmodels.cpp
@@ -139,8 +139,8 @@ void tst_QMail_ListModels::initTestCase()
     msg1.setFrom(QMailAddress("0404404040"));
     msg1.setTo(QMailAddress("0404040404"));
     msg1.setSubject("Where are you?");
-    msg1.setDate(QMailTimeStamp(QDate::currentDate().startOfDay()));
-    msg1.setReceivedDate(QMailTimeStamp(QDate::currentDate().startOfDay()));
+    msg1.setDate(QMailTimeStamp(QDateTime(QDate::currentDate())));
+    msg1.setReceivedDate(QMailTimeStamp(QDateTime(QDate::currentDate())));
     msg1.setStatus(QMailMessage::Incoming, true);
     msg1.setStatus(QMailMessage::New, false);
     msg1.setStatus(QMailMessage::Read, true);
@@ -158,8 +158,8 @@ void tst_QMail_ListModels::initTestCase()
     msg2.setTo(QMailAddress("old@example.org"));
     msg2.setCc(QMailAddressList() << QMailAddress("anotherguy@example.org"));
     msg2.setSubject("email message test");
-    msg2.setDate(QMailTimeStamp(QDate::currentDate().startOfDay()));
-    msg2.setReceivedDate(QMailTimeStamp(QDate::currentDate().startOfDay()));
+    msg2.setDate(QMailTimeStamp(QDateTime(QDate::currentDate())));
+    msg2.setReceivedDate(QMailTimeStamp(QDateTime(QDate::currentDate())));
     msg2.setStatus(QMailMessage::Incoming, true);
     msg2.setStatus(QMailMessage::New, true);
     msg2.setStatus(QMailMessage::Read, false);
diff --git a/tests/tst_qmail_sortkeys/tst_qmail_sortkeys.cpp b/tests/tst_qmail_sortkeys/tst_qmail_sortkeys.cpp
index 399c4968..4ab02483 100644
--- a/tests/tst_qmail_sortkeys/tst_qmail_sortkeys.cpp
+++ b/tests/tst_qmail_sortkeys/tst_qmail_sortkeys.cpp
@@ -261,8 +261,8 @@ void tst_QMail_SortKeys::initTestCase()
         message.setFrom(QMailAddress("0404404040"));
         message.setTo(QMailAddress("0404040404"));
         message.setSubject("Where are you?");
-        message.setDate(QMailTimeStamp(QDate::currentDate().startOfDay()));
-        message.setReceivedDate(QMailTimeStamp(QDate::currentDate().startOfDay()));
+        message.setDate(QMailTimeStamp(QDateTime(QDate::currentDate())));
+        message.setReceivedDate(QMailTimeStamp(QDateTime(QDate::currentDate())));
         message.setStatus(QMailMessage::Incoming, true);
         message.setStatus(QMailMessage::New, false);
         message.setStatus(QMailMessage::Read, true);
@@ -285,8 +285,8 @@ void tst_QMail_SortKeys::initTestCase()
         message.setFrom(QMailAddress("account2@example.org"));
         message.setTo(QMailAddress("account1@example.org"));
         message.setSubject("inboxMessage1");
-        message.setDate(QMailTimeStamp(QDate::currentDate().startOfDay()));
-        message.setReceivedDate(QMailTimeStamp(QDate::currentDate().startOfDay()));
+        message.setDate(QMailTimeStamp(QDateTime(QDate::currentDate())));
+        message.setReceivedDate(QMailTimeStamp(QDateTime(QDate::currentDate())));
         message.setStatus(QMailMessage::Incoming, true);
         message.setStatus(QMailMessage::New, true);
         message.setStatus(QMailMessage::Read, false);
@@ -309,8 +309,8 @@ void tst_QMail_SortKeys::initTestCase()
         message.setFrom(QMailAddress("account1@example.org"));
         message.setTo(QMailAddress("fred@example.net"));
         message.setSubject("archivedMessage1");
-        message.setDate(QMailTimeStamp(QDate::currentDate().addDays(-1).startOfDay()));
-        message.setReceivedDate(QMailTimeStamp(QDate::currentDate().addDays(-1).startOfDay()));
+        message.setDate(QMailTimeStamp(QDateTime(QDate::currentDate().addDays(-1))));
+        message.setReceivedDate(QMailTimeStamp(QDateTime(QDate::currentDate().addDays(-1))));
         message.setStatus(QMailMessage::Outgoing, true);
         message.setStatus(QMailMessage::New, false);
         message.setStatus(QMailMessage::Sent, true);
@@ -338,8 +338,8 @@ void tst_QMail_SortKeys::initTestCase()
         message.setFrom(QMailAddress("account1@example.org"));
         message.setTo(QMailAddress("account2@example.org"));
         message.setSubject("Fwd:inboxMessage2");
-        message.setDate(QMailTimeStamp(QDate::currentDate().startOfDay()));
-        message.setReceivedDate(QMailTimeStamp(QDate::currentDate().startOfDay()));
+        message.setDate(QMailTimeStamp(QDateTime(QDate::currentDate())));
+        message.setReceivedDate(QMailTimeStamp(QDateTime(QDate::currentDate())));
         message.setStatus(QMailMessage::Incoming, true);
         message.setStatus(QMailMessage::New, true);
         message.setStatus(QMailMessage::Read, true);
@@ -365,8 +365,8 @@ void tst_QMail_SortKeys::initTestCase()
         message.setFrom(QMailAddress("fred@example.net"));
         message.setTo(QMailAddressList() << QMailAddress("account2@example.org") << QMailAddress("testing@test"));
         message.setSubject("Re:savedMessage2");
-        message.setDate(QMailTimeStamp(QDate::currentDate().addDays(-7).startOfDay()));
-        message.setReceivedDate(QMailTimeStamp(QDate::currentDate().addDays(-7).startOfDay()));
+        message.setDate(QMailTimeStamp(QDateTime(QDate::currentDate().addDays(-7))));
+        message.setReceivedDate(QMailTimeStamp(QDateTime(QDate::currentDate().addDays(-7))));
         message.setStatus(QMailMessage::Incoming, true);
         message.setStatus(QMailMessage::New, false);
         message.setStatus(QMailMessage::Read, false);
diff --git a/tests/tst_qmaildisconnected/tst_qmaildisconnected.cpp b/tests/tst_qmaildisconnected/tst_qmaildisconnected.cpp
index dbdcb82c..b6fa02f0 100644
--- a/tests/tst_qmaildisconnected/tst_qmaildisconnected.cpp
+++ b/tests/tst_qmaildisconnected/tst_qmaildisconnected.cpp
@@ -260,8 +260,8 @@ void tst_QMailDisconnected::initTestCase()
         message.setFrom(QMailAddress("0404404040"));
         message.setTo(QMailAddress("0404040404"));
         message.setSubject("Where are you?");
-        message.setDate(QMailTimeStamp(QDate::currentDate().startOfDay()));
-        message.setReceivedDate(QMailTimeStamp(QDate::currentDate().startOfDay()));
+        message.setDate(QMailTimeStamp(QDateTime(QDate::currentDate())));
+        message.setReceivedDate(QMailTimeStamp(QDateTime(QDate::currentDate())));
         message.setStatus(QMailMessage::Incoming, true);
         message.setStatus(QMailMessage::New, false);
         message.setStatus(QMailMessage::Read, true);
@@ -284,8 +284,8 @@ void tst_QMailDisconnected::initTestCase()
         message.setFrom(QMailAddress("account2@example.org"));
         message.setTo(QMailAddress("account1@example.org"));
         message.setSubject("inboxMessage1");
-        message.setDate(QMailTimeStamp(QDate::currentDate().startOfDay()));
-        message.setReceivedDate(QMailTimeStamp(QDate::currentDate().startOfDay()));
+        message.setDate(QMailTimeStamp(QDateTime(QDate::currentDate())));
+        message.setReceivedDate(QMailTimeStamp(QDateTime(QDate::currentDate())));
         message.setStatus(QMailMessage::Incoming, true);
         message.setStatus(QMailMessage::New, true);
         message.setStatus(QMailMessage::Read, false);
@@ -308,8 +308,8 @@ void tst_QMailDisconnected::initTestCase()
         message.setFrom(QMailAddress("account1@example.org"));
         message.setTo(QMailAddress("fred@example.net"));
         message.setSubject("archivedMessage1");
-        message.setDate(QMailTimeStamp(QDate::currentDate().addDays(-1).startOfDay()));
-        message.setReceivedDate(QMailTimeStamp(QDate::currentDate().addDays(-1).startOfDay()));
+        message.setDate(QMailTimeStamp(QDateTime(QDate::currentDate().addDays(-1))));
+        message.setReceivedDate(QMailTimeStamp(QDateTime(QDate::currentDate().addDays(-1))));
         message.setStatus(QMailMessage::Outgoing, true);
         message.setStatus(QMailMessage::New, false);
         message.setStatus(QMailMessage::Sent, true);
@@ -337,8 +337,8 @@ void tst_QMailDisconnected::initTestCase()
         message.setFrom(QMailAddress("account1@example.org"));
         message.setTo(QMailAddress("account2@example.org"));
         message.setSubject("Fwd:inboxMessage2");
-        message.setDate(QMailTimeStamp(QDate::currentDate().startOfDay()));
-        message.setReceivedDate(QMailTimeStamp(QDate::currentDate().startOfDay()));
+        message.setDate(QMailTimeStamp(QDateTime(QDate::currentDate())));
+        message.setReceivedDate(QMailTimeStamp(QDateTime(QDate::currentDate())));
         message.setStatus(QMailMessage::Incoming, true);
         message.setStatus(QMailMessage::New, true);
         message.setStatus(QMailMessage::Read, true);
@@ -364,8 +364,8 @@ void tst_QMailDisconnected::initTestCase()
         message.setFrom(QMailAddress("fred@example.net"));
         message.setTo(QMailAddressList() << QMailAddress("account2@example.org") << QMailAddress("testing@test"));
         message.setSubject("Re:savedMessage2");
-        message.setDate(QMailTimeStamp(QDate::currentDate().addDays(-7).startOfDay()));
-        message.setReceivedDate(QMailTimeStamp(QDate::currentDate().addDays(-7).startOfDay()));
+        message.setDate(QMailTimeStamp(QDateTime(QDate::currentDate().addDays(-7))));
+        message.setReceivedDate(QMailTimeStamp(QDateTime(QDate::currentDate().addDays(-7))));
         message.setStatus(QMailMessage::Incoming, true);
         message.setStatus(QMailMessage::New, false);
         message.setStatus(QMailMessage::Read, false);
@@ -406,7 +406,7 @@ void tst_QMailDisconnected::test_qmaildisconnected()
     QMailMessage dstMsg;
     QMailDisconnected::copyPreviousFolder(QMailMessage(savedMessage2), &dstMsg);
 
-    QMap<QMailFolderId, QMailMessageIdList> map = QMailDisconnected::restoreMap(allMessages.values());
+    QMap<QMailFolderId, QMailMessageIdList> map = QMailDisconnected::restoreMap(allMessages.toList());
 
     QMailDisconnected::copyToFolder(QMailMessageIdList() << inboxMessage1, archivedId1);
     QMailDisconnected::copyToStandardFolder(QMailMessageIdList() << inboxMessage1, QMailFolder::JunkFolder);
diff --git a/tests/tst_qmailmessageset/tst_qmailmessageset.cpp b/tests/tst_qmailmessageset/tst_qmailmessageset.cpp
index c63212d8..13854942 100644
--- a/tests/tst_qmailmessageset/tst_qmailmessageset.cpp
+++ b/tests/tst_qmailmessageset/tst_qmailmessageset.cpp
@@ -267,8 +267,8 @@ void tst_QMailMessageSet::initTestCase()
         message.setFrom(QMailAddress("0404404040"));
         message.setTo(QMailAddress("0404040404"));
         message.setSubject("Where are you?");
-        message.setDate(QMailTimeStamp(QDate::currentDate().startOfDay()));
-        message.setReceivedDate(QMailTimeStamp(QDate::currentDate().startOfDay()));
+        message.setDate(QMailTimeStamp(QDateTime(QDate::currentDate())));
+        message.setReceivedDate(QMailTimeStamp(QDateTime(QDate::currentDate())));
         message.setStatus(QMailMessage::Incoming, true);
         message.setStatus(QMailMessage::New, false);
         message.setStatus(QMailMessage::Read, true);
@@ -291,8 +291,8 @@ void tst_QMailMessageSet::initTestCase()
         message.setFrom(QMailAddress("account2@example.org"));
         message.setTo(QMailAddress("account1@example.org"));
         message.setSubject("inboxMessage1");
-        message.setDate(QMailTimeStamp(QDate::currentDate().startOfDay()));
-        message.setReceivedDate(QMailTimeStamp(QDate::currentDate().startOfDay()));
+        message.setDate(QMailTimeStamp(QDateTime(QDate::currentDate())));
+        message.setReceivedDate(QMailTimeStamp(QDateTime(QDate::currentDate())));
         message.setStatus(QMailMessage::Incoming, true);
         message.setStatus(QMailMessage::New, true);
         message.setStatus(QMailMessage::Read, false);
@@ -315,8 +315,8 @@ void tst_QMailMessageSet::initTestCase()
         message.setFrom(QMailAddress("account1@example.org"));
         message.setTo(QMailAddress("fred@example.net"));
         message.setSubject("archivedMessage1");
-        message.setDate(QMailTimeStamp(QDate::currentDate().addDays(-1).startOfDay()));
-        message.setReceivedDate(QMailTimeStamp(QDate::currentDate().addDays(-1).startOfDay()));
+        message.setDate(QMailTimeStamp(QDateTime(QDate::currentDate().addDays(-1))));
+        message.setReceivedDate(QMailTimeStamp(QDateTime(QDate::currentDate().addDays(-1))));
         message.setStatus(QMailMessage::Outgoing, true);
         message.setStatus(QMailMessage::New, false);
         message.setStatus(QMailMessage::Sent, true);
@@ -344,8 +344,8 @@ void tst_QMailMessageSet::initTestCase()
         message.setFrom(QMailAddress("account1@example.org"));
         message.setTo(QMailAddress("account2@example.org"));
         message.setSubject("Fwd:inboxMessage2");
-        message.setDate(QMailTimeStamp(QDate::currentDate().startOfDay()));
-        message.setReceivedDate(QMailTimeStamp(QDate::currentDate().startOfDay()));
+        message.setDate(QMailTimeStamp(QDateTime(QDate::currentDate())));
+        message.setReceivedDate(QMailTimeStamp(QDateTime(QDate::currentDate())));
         message.setStatus(QMailMessage::Incoming, true);
         message.setStatus(QMailMessage::New, true);
         message.setStatus(QMailMessage::Read, true);
@@ -371,8 +371,8 @@ void tst_QMailMessageSet::initTestCase()
         message.setFrom(QMailAddress("fred@example.net"));
         message.setTo(QMailAddressList() << QMailAddress("account2@example.org") << QMailAddress("testing@test"));
         message.setSubject("Re:savedMessage2");
-        message.setDate(QMailTimeStamp(QDate::currentDate().addDays(-7).startOfDay()));
-        message.setReceivedDate(QMailTimeStamp(QDate::currentDate().addDays(-7).startOfDay()));
+        message.setDate(QMailTimeStamp(QDateTime(QDate::currentDate().addDays(-7))));
+        message.setReceivedDate(QMailTimeStamp(QDateTime(QDate::currentDate().addDays(-7))));
         message.setStatus(QMailMessage::Incoming, true);
         message.setStatus(QMailMessage::New, false);
         message.setStatus(QMailMessage::Read, false);
diff --git a/tests/tst_qmailserviceaction/tst_qmailserviceaction.cpp b/tests/tst_qmailserviceaction/tst_qmailserviceaction.cpp
index aa82f476..efcfad4a 100644
--- a/tests/tst_qmailserviceaction/tst_qmailserviceaction.cpp
+++ b/tests/tst_qmailserviceaction/tst_qmailserviceaction.cpp
@@ -265,8 +265,8 @@ void tst_QMailServiceAction::initTestCase()
         message.setFrom(QMailAddress("0404404040"));
         message.setTo(QMailAddress("0404040404"));
         message.setSubject("Where are you?");
-        message.setDate(QMailTimeStamp(QDate::currentDate().startOfDay()));
-        message.setReceivedDate(QMailTimeStamp(QDate::currentDate().startOfDay()));
+        message.setDate(QMailTimeStamp(QDateTime(QDate::currentDate())));
+        message.setReceivedDate(QMailTimeStamp(QDateTime(QDate::currentDate())));
         message.setStatus(QMailMessage::Incoming, true);
         message.setStatus(QMailMessage::New, false);
         message.setStatus(QMailMessage::Read, true);
@@ -289,8 +289,8 @@ void tst_QMailServiceAction::initTestCase()
         message.setFrom(QMailAddress("account2@example.org"));
         message.setTo(QMailAddress("account1@example.org"));
         message.setSubject("inboxMessage1");
-        message.setDate(QMailTimeStamp(QDate::currentDate().startOfDay()));
-        message.setReceivedDate(QMailTimeStamp(QDate::currentDate().startOfDay()));
+        message.setDate(QMailTimeStamp(QDateTime(QDate::currentDate())));
+        message.setReceivedDate(QMailTimeStamp(QDateTime(QDate::currentDate())));
         message.setStatus(QMailMessage::Incoming, true);
         message.setStatus(QMailMessage::New, true);
         message.setStatus(QMailMessage::Read, false);
@@ -313,8 +313,8 @@ void tst_QMailServiceAction::initTestCase()
         message.setFrom(QMailAddress("account1@example.org"));
         message.setTo(QMailAddress("fred@example.net"));
         message.setSubject("archivedMessage1");
-        message.setDate(QMailTimeStamp(QDate::currentDate().addDays(-1).startOfDay()));
-        message.setReceivedDate(QMailTimeStamp(QDate::currentDate().addDays(-1).startOfDay()));
+        message.setDate(QMailTimeStamp(QDateTime(QDate::currentDate().addDays(-1))));
+        message.setReceivedDate(QMailTimeStamp(QDateTime(QDate::currentDate().addDays(-1))));
         message.setStatus(QMailMessage::Outgoing, true);
         message.setStatus(QMailMessage::New, false);
         message.setStatus(QMailMessage::Sent, true);
@@ -342,8 +342,8 @@ void tst_QMailServiceAction::initTestCase()
         message.setFrom(QMailAddress("account1@example.org"));
         message.setTo(QMailAddress("account2@example.org"));
         message.setSubject("Fwd:inboxMessage2");
-        message.setDate(QMailTimeStamp(QDate::currentDate().startOfDay()));
-        message.setReceivedDate(QMailTimeStamp(QDate::currentDate().startOfDay()));
+        message.setDate(QMailTimeStamp(QDateTime(QDate::currentDate())));
+        message.setReceivedDate(QMailTimeStamp(QDateTime(QDate::currentDate())));
         message.setStatus(QMailMessage::Incoming, true);
         message.setStatus(QMailMessage::New, true);
         message.setStatus(QMailMessage::Read, true);
@@ -369,8 +369,8 @@ void tst_QMailServiceAction::initTestCase()
         message.setFrom(QMailAddress("fred@example.net"));
         message.setTo(QMailAddressList() << QMailAddress("account2@example.org") << QMailAddress("testing@test"));
         message.setSubject("Re:savedMessage2");
-        message.setDate(QMailTimeStamp(QDate::currentDate().addDays(-7).startOfDay()));
-        message.setReceivedDate(QMailTimeStamp(QDate::currentDate().addDays(-7).startOfDay()));
+        message.setDate(QMailTimeStamp(QDateTime(QDate::currentDate().addDays(-7))));
+        message.setReceivedDate(QMailTimeStamp(QDateTime(QDate::currentDate().addDays(-7))));
         message.setStatus(QMailMessage::Incoming, true);
         message.setStatus(QMailMessage::New, false);
         message.setStatus(QMailMessage::Read, false);
@@ -407,7 +407,7 @@ void tst_QMailServiceAction::test_retrievalaction()
     uint min = 10240u;
     action.retrieveFolderList(accountId1, inboxId1);
     action.retrieveMessageList(accountId2, inboxId2);
-    action.retrieveMessages(allMessages.values());
+    action.retrieveMessages(allMessages.toList());
     action.retrieveMessageRange(inboxMessage1, min);
 
     action.exportUpdates(accountId1);
diff --git a/tests/tst_qmailstorageaction/tst_qmailstorageaction.cpp b/tests/tst_qmailstorageaction/tst_qmailstorageaction.cpp
index 9a8497a0..a397bc3d 100644
--- a/tests/tst_qmailstorageaction/tst_qmailstorageaction.cpp
+++ b/tests/tst_qmailstorageaction/tst_qmailstorageaction.cpp
@@ -341,8 +341,8 @@ void tst_QMailStorageAction::initTestCase()
         message.setFrom(QMailAddress("0404404040"));
         message.setTo(QMailAddress("0404040404"));
         message.setSubject("Where are you?");
-        message.setDate(QMailTimeStamp(QDate::currentDate().startOfDay()));
-        message.setReceivedDate(QMailTimeStamp(QDate::currentDate().startOfDay()));
+        message.setDate(QMailTimeStamp(QDateTime(QDate::currentDate())));
+        message.setReceivedDate(QMailTimeStamp(QDateTime(QDate::currentDate())));
         message.setStatus(QMailMessage::Incoming, true);
         message.setStatus(QMailMessage::New, false);
         message.setStatus(QMailMessage::Read, true);
@@ -365,8 +365,8 @@ void tst_QMailStorageAction::initTestCase()
         message.setFrom(QMailAddress("account2@example.org"));
         message.setTo(QMailAddress("account1@example.org"));
         message.setSubject("inboxMessage1");
-        message.setDate(QMailTimeStamp(QDate::currentDate().startOfDay()));
-        message.setReceivedDate(QMailTimeStamp(QDate::currentDate().startOfDay()));
+        message.setDate(QMailTimeStamp(QDateTime(QDate::currentDate())));
+        message.setReceivedDate(QMailTimeStamp(QDateTime(QDate::currentDate())));
         message.setStatus(QMailMessage::Incoming, true);
         message.setStatus(QMailMessage::New, true);
         message.setStatus(QMailMessage::Read, false);
@@ -389,8 +389,8 @@ void tst_QMailStorageAction::initTestCase()
         message.setFrom(QMailAddress("account1@example.org"));
         message.setTo(QMailAddress("fred@example.net"));
         message.setSubject("archivedMessage1");
-        message.setDate(QMailTimeStamp(QDate::currentDate().addDays(-1).startOfDay()));
-        message.setReceivedDate(QMailTimeStamp(QDate::currentDate().addDays(-1).startOfDay()));
+        message.setDate(QMailTimeStamp(QDateTime(QDate::currentDate().addDays(-1))));
+        message.setReceivedDate(QMailTimeStamp(QDateTime(QDate::currentDate().addDays(-1))));
         message.setStatus(QMailMessage::Outgoing, true);
         message.setStatus(QMailMessage::New, false);
         message.setStatus(QMailMessage::Sent, true);
@@ -418,8 +418,8 @@ void tst_QMailStorageAction::initTestCase()
         message.setFrom(QMailAddress("account1@example.org"));
         message.setTo(QMailAddress("account2@example.org"));
         message.setSubject("Fwd:inboxMessage2");
-        message.setDate(QMailTimeStamp(QDate::currentDate().startOfDay()));
-        message.setReceivedDate(QMailTimeStamp(QDate::currentDate().startOfDay()));
+        message.setDate(QMailTimeStamp(QDateTime(QDate::currentDate())));
+        message.setReceivedDate(QMailTimeStamp(QDateTime(QDate::currentDate())));
         message.setStatus(QMailMessage::Incoming, true);
         message.setStatus(QMailMessage::New, true);
         message.setStatus(QMailMessage::Read, true);
@@ -445,8 +445,8 @@ void tst_QMailStorageAction::initTestCase()
         message.setFrom(QMailAddress("fred@example.net"));
         message.setTo(QMailAddressList() << QMailAddress("account2@example.org") << QMailAddress("testing@test"));
         message.setSubject("Re:savedMessage2");
-        message.setDate(QMailTimeStamp(QDate::currentDate().addDays(-7).startOfDay()));
-        message.setReceivedDate(QMailTimeStamp(QDate::currentDate().addDays(-7).startOfDay()));
+        message.setDate(QMailTimeStamp(QDateTime(QDate::currentDate().addDays(-7))));
+        message.setReceivedDate(QMailTimeStamp(QDateTime(QDate::currentDate().addDays(-7))));
         message.setStatus(QMailMessage::Incoming, true);
         message.setStatus(QMailMessage::New, false);
         message.setStatus(QMailMessage::Read, false);
@@ -489,8 +489,8 @@ void tst_QMailStorageAction::test_storageaction_add()
     message.setFrom(QMailAddress("wilma@example.net"));
     message.setTo(QMailAddressList() << QMailAddress("account2@example.org") << QMailAddress("testing@test"));
     message.setSubject("Simple test message");
-    message.setDate(QMailTimeStamp(QDate::currentDate().addDays(-100).startOfDay()));
-    message.setReceivedDate(QMailTimeStamp(QDate::currentDate().addDays(-100).startOfDay()));
+    message.setDate(QMailTimeStamp(QDateTime(QDate::currentDate().addDays(-100))));
+    message.setReceivedDate(QMailTimeStamp(QDateTime(QDate::currentDate().addDays(-100))));
     message.setStatus(QMailMessage::Incoming, true);
     message.setStatus(QMailMessage::New, false);
     message.setStatus(QMailMessage::Read, false);
@@ -741,8 +741,8 @@ void tst_QMailStorageAction::test_storageaction_rollBackUpdates()
     message.setFrom(QMailAddress("barney@example.net"));
     message.setTo(QMailAddressList() << QMailAddress("account3@example.org") << QMailAddress("testing@test"));
     message.setSubject("Rollback test message");
-    message.setDate(QMailTimeStamp(QDate::currentDate().addDays(-98).startOfDay()));
-    message.setReceivedDate(QMailTimeStamp(QDate::currentDate().addDays(-98).startOfDay()));
+    message.setDate(QMailTimeStamp(QDateTime(QDate::currentDate().addDays(-98))));
+    message.setReceivedDate(QMailTimeStamp(QDateTime(QDate::currentDate().addDays(-98))));
     message.setStatus(QMailMessage::Incoming, true);
     message.setStatus(QMailMessage::New, false);
     message.setStatus(QMailMessage::Read, false);
@@ -837,8 +837,8 @@ void tst_QMailStorageAction::test_storageaction_discardMessages()
     message.setFrom(QMailAddress("barney@example.net"));
     message.setTo(QMailAddressList() << QMailAddress("account2@example.org") << QMailAddress("testing@test"));
     message.setSubject("Another test message");
-    message.setDate(QMailTimeStamp(QDate::currentDate().addDays(-99).startOfDay()));
-    message.setReceivedDate(QMailTimeStamp(QDate::currentDate().addDays(-99).startOfDay()));
+    message.setDate(QMailTimeStamp(QDateTime(QDate::currentDate().addDays(-99))));
+    message.setReceivedDate(QMailTimeStamp(QDateTime(QDate::currentDate().addDays(-99))));
     message.setStatus(QMailMessage::Incoming, true);
     message.setStatus(QMailMessage::New, false);
     message.setStatus(QMailMessage::Read, false);
diff --git a/tests/tst_qmailstore/tst_qmailstore.cpp b/tests/tst_qmailstore/tst_qmailstore.cpp
index 287befa7..94845204 100644
--- a/tests/tst_qmailstore/tst_qmailstore.cpp
+++ b/tests/tst_qmailstore/tst_qmailstore.cpp
@@ -1776,8 +1776,8 @@ void tst_QMailStore::message()
     msg2.setParentAccountId(account2.id());
     msg2.setParentFolderId(QMailFolderId::LocalStorageFolderId);
     msg2.setSubject("email message test");
-    msg2.setDate(QMailTimeStamp(QDate::currentDate().startOfDay()));
-    msg2.setReceivedDate(QMailTimeStamp(QDate::currentDate().startOfDay()));
+    msg2.setDate(QMailTimeStamp(QDateTime(QDate::currentDate())));
+    msg2.setReceivedDate(QMailTimeStamp(QDateTime(QDate::currentDate())));
     msg2.setStatus(QMailMessage::Incoming, true);
     msg2.setStatus(QMailMessage::New, true);
     msg2.setStatus(QMailMessage::Read, false);
diff --git a/tests/tst_qmailstorekeys/tst_qmailstorekeys.cpp b/tests/tst_qmailstorekeys/tst_qmailstorekeys.cpp
index 6891bc78..4a311fee 100644
--- a/tests/tst_qmailstorekeys/tst_qmailstorekeys.cpp
+++ b/tests/tst_qmailstorekeys/tst_qmailstorekeys.cpp
@@ -98,8 +98,7 @@ private:
     // We only want to compare sets, disregarding ordering
     const QSet<QMailAccountId> accountSet(const QMailAccountKey &key) const
     {
-        const QMailAccountIdList &accounts(QMailStore::instance()->queryAccounts(key));
-        return QSet<QMailAccountId>(accounts.constBegin(), accounts.constEnd());
+        return QMailStore::instance()->queryAccounts(key).toSet();
     }
 
     QSet<QMailAccountId> accountSet() const
@@ -109,8 +108,7 @@ private:
 
     const QSet<QMailFolderId> folderSet(const QMailFolderKey &key) const
     {
-        const QMailFolderIdList &folders(QMailStore::instance()->queryFolders(key));
-        return QSet<QMailFolderId>(folders.constBegin(), folders.constEnd());
+        return QMailStore::instance()->queryFolders(key).toSet();
     }
 
     QSet<QMailFolderId> folderSet() const
@@ -120,8 +118,7 @@ private:
 
     const QSet<QMailMessageId> messageSet(const QMailMessageKey &key) const
     {
-        const QMailMessageIdList &messages(QMailStore::instance()->queryMessages(key));
-        return QSet<QMailMessageId>(messages.constBegin(), messages.constEnd());
+        return QMailStore::instance()->queryMessages(key).toSet();
     }
 
     QSet<QMailMessageId> messageSet() const
@@ -343,8 +340,8 @@ void tst_QMailStoreKeys::initTestCase()
         message.setFrom(QMailAddress("0404404040"));
         message.setTo(QMailAddress("0404040404"));
         message.setSubject("Where are you?");
-        message.setDate(QMailTimeStamp(QDate::currentDate().startOfDay()));
-        message.setReceivedDate(QMailTimeStamp(QDate::currentDate().startOfDay()));
+        message.setDate(QMailTimeStamp(QDateTime(QDate::currentDate())));
+        message.setReceivedDate(QMailTimeStamp(QDateTime(QDate::currentDate())));
         message.setStatus(QMailMessage::Incoming, true);
         message.setStatus(QMailMessage::New, false);
         message.setStatus(QMailMessage::Read, true);
@@ -367,8 +364,8 @@ void tst_QMailStoreKeys::initTestCase()
         message.setFrom(QMailAddress("account2@example.org"));
         message.setTo(QMailAddress("account1@example.org"));
         message.setSubject("inboxMessage1");
-        message.setDate(QMailTimeStamp(QDate::currentDate().startOfDay()));
-        message.setReceivedDate(QMailTimeStamp(QDate::currentDate().startOfDay()));
+        message.setDate(QMailTimeStamp(QDateTime(QDate::currentDate())));
+        message.setReceivedDate(QMailTimeStamp(QDateTime(QDate::currentDate())));
         message.setStatus(QMailMessage::Incoming, true);
         message.setStatus(QMailMessage::New, true);
         message.setStatus(QMailMessage::Read, false);
@@ -391,8 +388,8 @@ void tst_QMailStoreKeys::initTestCase()
         message.setFrom(QMailAddress("account1@example.org"));
         message.setTo(QMailAddress("fred@example.net"));
         message.setSubject("archivedMessage1");
-        message.setDate(QMailTimeStamp(QDate::currentDate().addDays(-1).startOfDay()));
-        message.setReceivedDate(QMailTimeStamp(QDate::currentDate().addDays(-1).startOfDay()));
+        message.setDate(QMailTimeStamp(QDateTime(QDate::currentDate().addDays(-1))));
+        message.setReceivedDate(QMailTimeStamp(QDateTime(QDate::currentDate().addDays(-1))));
         message.setStatus(QMailMessage::Outgoing, true);
         message.setStatus(QMailMessage::New, false);
         message.setStatus(QMailMessage::Sent, true);
@@ -420,8 +417,8 @@ void tst_QMailStoreKeys::initTestCase()
         message.setFrom(QMailAddress("account1@example.org"));
         message.setTo(QMailAddress("account2@example.org"));
         message.setSubject("Fwd:inboxMessage2");
-        message.setDate(QMailTimeStamp(QDate::currentDate().startOfDay()));
-        message.setReceivedDate(QMailTimeStamp(QDate::currentDate().startOfDay()));
+        message.setDate(QMailTimeStamp(QDateTime(QDate::currentDate())));
+        message.setReceivedDate(QMailTimeStamp(QDateTime(QDate::currentDate())));
         message.setStatus(QMailMessage::Incoming, true);
         message.setStatus(QMailMessage::New, true);
         message.setStatus(QMailMessage::Read, true);
@@ -447,8 +444,8 @@ void tst_QMailStoreKeys::initTestCase()
         message.setFrom(QMailAddress("fred@example.net"));
         message.setTo(QMailAddressList() << QMailAddress("account2@example.org") << QMailAddress("testing@test"));
         message.setSubject("Re:savedMessage2");
-        message.setDate(QMailTimeStamp(QDate::currentDate().addDays(-7).startOfDay()));
-        message.setReceivedDate(QMailTimeStamp(QDate::currentDate().addDays(-7).startOfDay()));
+        message.setDate(QMailTimeStamp(QDateTime(QDate::currentDate().addDays(-7))));
+        message.setReceivedDate(QMailTimeStamp(QDateTime(QDate::currentDate().addDays(-7))));
         message.setStatus(QMailMessage::Incoming, true);
         message.setStatus(QMailMessage::New, false);
         message.setStatus(QMailMessage::Read, false);
@@ -570,9 +567,8 @@ void tst_QMailStoreKeys::accountId()
     QCOMPARE(accountSet(~QMailAccountKey::id(QMailAccountId(), NotEqual)), noAccounts);
 
     // List inclusion
-    const QMailAccountIdList accounts(allAccounts.constBegin(), allAccounts.constEnd());
-    QCOMPARE(accountSet(QMailAccountKey::id(accounts)), allAccounts);
-    QCOMPARE(accountSet(~QMailAccountKey::id(accounts)), noAccounts);
+    QCOMPARE(accountSet(QMailAccountKey::id(allAccounts.toList())), allAccounts);
+    QCOMPARE(accountSet(~QMailAccountKey::id(allAccounts.toList())), noAccounts);
     QCOMPARE(accountSet(QMailAccountKey::id(QMailAccountIdList() << accountId1)), accountSet() << accountId1);
     QCOMPARE(accountSet(~QMailAccountKey::id(QMailAccountIdList() << accountId1)), accountSet() << accountId2 << accountId3 << accountId4);
     QCOMPARE(accountSet(QMailAccountKey::id(QMailAccountIdList() << accountId2)), accountSet() << accountId2);
@@ -581,8 +577,8 @@ void tst_QMailStoreKeys::accountId()
     QCOMPARE(accountSet(~QMailAccountKey::id(QMailAccountIdList() << accountId1 << accountId2)), accountSet() << accountId3 << accountId4);
 
     // List exclusion
-    QCOMPARE(accountSet(QMailAccountKey::id(accounts, Excludes)), noAccounts);
-    QCOMPARE(accountSet(~QMailAccountKey::id(accounts, Excludes)), allAccounts);
+    QCOMPARE(accountSet(QMailAccountKey::id(allAccounts.toList(), Excludes)), noAccounts);
+    QCOMPARE(accountSet(~QMailAccountKey::id(allAccounts.toList(), Excludes)), allAccounts);
     QCOMPARE(accountSet(QMailAccountKey::id(QMailAccountIdList() << accountId1, Excludes)), accountSet() << accountId2 << accountId3 << accountId4);
     QCOMPARE(accountSet(~QMailAccountKey::id(QMailAccountIdList() << accountId1, Excludes)), accountSet() << accountId1);
     QCOMPARE(accountSet(QMailAccountKey::id(QMailAccountIdList() << accountId2, Excludes)), accountSet() << accountId1 << accountId3 << accountId4);
@@ -842,9 +838,8 @@ void tst_QMailStoreKeys::folderId()
     QCOMPARE(folderSet(~QMailFolderKey::id(QMailFolderId(), NotEqual)), noFolders);
 
     // List inclusion
-    const QMailFolderIdList folders(allFolders.constBegin(), allFolders.constEnd());
-    QCOMPARE(folderSet(QMailFolderKey::id(folders)), allFolders);
-    QCOMPARE(folderSet(~QMailFolderKey::id(folders)), standardFolders);
+    QCOMPARE(folderSet(QMailFolderKey::id(allFolders.toList())), allFolders);
+    QCOMPARE(folderSet(~QMailFolderKey::id(allFolders.toList())), standardFolders);
     QCOMPARE(folderSet(QMailFolderKey::id(QMailFolderIdList() << inboxId1)), folderSet() << inboxId1);
     QCOMPARE(folderSet(~QMailFolderKey::id(QMailFolderIdList() << inboxId1)), standardFolders + folderSet() << savedId1 << archivedId1 << inboxId2 << savedId2 << archivedId2);
     QCOMPARE(folderSet(QMailFolderKey::id(QMailFolderIdList() << archivedId2)), folderSet() << archivedId2);
@@ -853,8 +848,8 @@ void tst_QMailStoreKeys::folderId()
     QCOMPARE(folderSet(~QMailFolderKey::id(QMailFolderIdList() << inboxId1 << archivedId2)), standardFolders + folderSet() << savedId1 << archivedId1 << inboxId2 << savedId2);
 
     // List exclusion
-    QCOMPARE(folderSet(QMailFolderKey::id(folders, Excludes)), standardFolders);
-    QCOMPARE(folderSet(~QMailFolderKey::id(folders, Excludes)), allFolders);
+    QCOMPARE(folderSet(QMailFolderKey::id(allFolders.toList(), Excludes)), standardFolders);
+    QCOMPARE(folderSet(~QMailFolderKey::id(allFolders.toList(), Excludes)), allFolders);
     QCOMPARE(folderSet(QMailFolderKey::id(QMailFolderIdList() << inboxId1, Excludes)), standardFolders + folderSet() << savedId1 << archivedId1 << inboxId2 << savedId2 << archivedId2);
     QCOMPARE(folderSet(~QMailFolderKey::id(QMailFolderIdList() << inboxId1, Excludes)), folderSet() << inboxId1);
     QCOMPARE(folderSet(QMailFolderKey::id(QMailFolderIdList() << archivedId2, Excludes)), standardFolders + folderSet() << inboxId1 << savedId1 << archivedId1 << inboxId2 << savedId2);
@@ -1324,9 +1319,8 @@ void tst_QMailStoreKeys::messageId()
     QCOMPARE(messageSet(~QMailMessageKey::id(QMailMessageId(), NotEqual)), noMessages);
 
     // List inclusion
-    const QMailMessageIdList messages(allMessages.constBegin(), allMessages.constEnd());
-    QCOMPARE(messageSet(QMailMessageKey::id(messages)), allMessages);
-    QCOMPARE(messageSet(~QMailMessageKey::id(messages)), noMessages);
+    QCOMPARE(messageSet(QMailMessageKey::id(allMessages.toList())), allMessages);
+    QCOMPARE(messageSet(~QMailMessageKey::id(allMessages.toList())), noMessages);
     QCOMPARE(messageSet(QMailMessageKey::id(QMailMessageIdList() << smsMessage)), messageSet() << smsMessage);
     QCOMPARE(messageSet(~QMailMessageKey::id(QMailMessageIdList() << smsMessage)), allEmailMessages);
     QCOMPARE(messageSet(QMailMessageKey::id(QMailMessageIdList() << inboxMessage1)), messageSet() << inboxMessage1);
@@ -1335,8 +1329,8 @@ void tst_QMailStoreKeys::messageId()
     QCOMPARE(messageSet(~QMailMessageKey::id(QMailMessageIdList() << smsMessage << inboxMessage1)), messageSet() << archivedMessage1 << inboxMessage2 << savedMessage2);
 
     // List Exclusion
-    QCOMPARE(messageSet(QMailMessageKey::id(messages, Excludes)), noMessages);
-    QCOMPARE(messageSet(~QMailMessageKey::id(messages, Excludes)), allMessages);
+    QCOMPARE(messageSet(QMailMessageKey::id(allMessages.toList(), Excludes)), noMessages);
+    QCOMPARE(messageSet(~QMailMessageKey::id(allMessages.toList(), Excludes)), allMessages);
     QCOMPARE(messageSet(QMailMessageKey::id(QMailMessageIdList() << smsMessage, Excludes)), allEmailMessages);
     QCOMPARE(messageSet(~QMailMessageKey::id(QMailMessageIdList() << smsMessage, Excludes)), messageSet() << smsMessage);
     QCOMPARE(messageSet(QMailMessageKey::id(QMailMessageIdList() << inboxMessage1, Excludes)), messageSet() << smsMessage << archivedMessage1 << inboxMessage2 << savedMessage2);
@@ -1634,7 +1628,7 @@ void tst_QMailStoreKeys::messageSubject()
 
 void tst_QMailStoreKeys::messageTimeStamp()
 {
-    QDateTime today(QDate::currentDate().startOfDay()), yesterday(QDate::currentDate().addDays(-1).startOfDay()), lastWeek(QDate::currentDate().addDays(-7).startOfDay());
+    QDateTime today(QDate::currentDate()), yesterday(QDate::currentDate().addDays(-1)), lastWeek(QDate::currentDate().addDays(-7));
     today = today.toUTC();
     yesterday = yesterday.toUTC();
     lastWeek = lastWeek.toUTC();
@@ -1694,7 +1688,7 @@ void tst_QMailStoreKeys::messageTimeStamp()
 
 void tst_QMailStoreKeys::messageReceptionTimeStamp()
 {
-    QDateTime today(QDate::currentDate().startOfDay()), yesterday(QDate::currentDate().addDays(-1).startOfDay()), lastWeek(QDate::currentDate().addDays(-7).startOfDay());
+    QDateTime today(QDate::currentDate()), yesterday(QDate::currentDate().addDays(-1)), lastWeek(QDate::currentDate().addDays(-7));
     today = today.toUTC();
     yesterday = yesterday.toUTC();
     lastWeek = lastWeek.toUTC();
diff --git a/tests/tst_qmailthread/tst_qmailthread.cpp b/tests/tst_qmailthread/tst_qmailthread.cpp
index 2709c581..7cf80505 100644
--- a/tests/tst_qmailthread/tst_qmailthread.cpp
+++ b/tests/tst_qmailthread/tst_qmailthread.cpp
@@ -284,8 +284,8 @@ void tst_qmailthread::initTestCase()
         message.setFrom(QMailAddress("0404404040"));
         message.setTo(QMailAddress("0404040404"));
         message.setSubject("Where are you?");
-        message.setDate(QMailTimeStamp(QDate::currentDate().startOfDay()));
-        message.setReceivedDate(QMailTimeStamp(QDate::currentDate().startOfDay()));
+        message.setDate(QMailTimeStamp(QDateTime(QDate::currentDate())));
+        message.setReceivedDate(QMailTimeStamp(QDateTime(QDate::currentDate())));
         message.setStatus(QMailMessage::Incoming, true);
         message.setStatus(QMailMessage::New, false);
         message.setStatus(QMailMessage::Read, true);
@@ -308,8 +308,8 @@ void tst_qmailthread::initTestCase()
         message.setFrom(QMailAddress("account2@example.org"));
         message.setTo(QMailAddress("account1@example.org"));
         message.setSubject("inboxMessage1");
-        message.setDate(QMailTimeStamp(QDate::currentDate().startOfDay()));
-        message.setReceivedDate(QMailTimeStamp(QDate::currentDate().startOfDay()));
+        message.setDate(QMailTimeStamp(QDateTime(QDate::currentDate())));
+        message.setReceivedDate(QMailTimeStamp(QDateTime(QDate::currentDate())));
         message.setStatus(QMailMessage::Incoming, true);
         message.setStatus(QMailMessage::New, true);
         message.setStatus(QMailMessage::Read, false);
@@ -332,8 +332,8 @@ void tst_qmailthread::initTestCase()
         message.setFrom(QMailAddress("account1@example.org"));
         message.setTo(QMailAddress("fred@example.net"));
         message.setSubject("archivedMessage1");
-        message.setDate(QMailTimeStamp(QDate::currentDate().addDays(-1).startOfDay()));
-        message.setReceivedDate(QMailTimeStamp(QDate::currentDate().addDays(-1).startOfDay()));
+        message.setDate(QMailTimeStamp(QDateTime(QDate::currentDate().addDays(-1))));
+        message.setReceivedDate(QMailTimeStamp(QDateTime(QDate::currentDate().addDays(-1))));
         message.setStatus(QMailMessage::Outgoing, true);
         message.setStatus(QMailMessage::New, false);
         message.setStatus(QMailMessage::Sent, true);
@@ -361,8 +361,8 @@ void tst_qmailthread::initTestCase()
         message.setFrom(QMailAddress("account1@example.org"));
         message.setTo(QMailAddress("account2@example.org"));
         message.setSubject("Fwd:inboxMessage2");
-        message.setDate(QMailTimeStamp(QDate::currentDate().startOfDay()));
-        message.setReceivedDate(QMailTimeStamp(QDate::currentDate().startOfDay()));
+        message.setDate(QMailTimeStamp(QDateTime(QDate::currentDate())));
+        message.setReceivedDate(QMailTimeStamp(QDateTime(QDate::currentDate())));
         message.setStatus(QMailMessage::Incoming, true);
         message.setStatus(QMailMessage::New, true);
         message.setStatus(QMailMessage::Read, true);
@@ -388,8 +388,8 @@ void tst_qmailthread::initTestCase()
         message.setFrom(QMailAddress("fred@example.net"));
         message.setTo(QMailAddressList() << QMailAddress("account2@example.org") << QMailAddress("testing@test"));
         message.setSubject("Re:savedMessage2");
-        message.setDate(QMailTimeStamp(QDate::currentDate().addDays(-7).startOfDay()));
-        message.setReceivedDate(QMailTimeStamp(QDate::currentDate().addDays(-7).startOfDay()));
+        message.setDate(QMailTimeStamp(QDateTime(QDate::currentDate().addDays(-7))));
+        message.setReceivedDate(QMailTimeStamp(QDateTime(QDate::currentDate().addDays(-7))));
         message.setStatus(QMailMessage::Incoming, true);
         message.setStatus(QMailMessage::New, false);
         message.setStatus(QMailMessage::Read, false);
diff --git a/tests/tst_storagemanager/tst_storagemanager.cpp b/tests/tst_storagemanager/tst_storagemanager.cpp
index 48926590..ea9f846c 100644
--- a/tests/tst_storagemanager/tst_storagemanager.cpp
+++ b/tests/tst_storagemanager/tst_storagemanager.cpp
@@ -153,8 +153,8 @@ void tst_StorageManager::initTestCase()
     msg1.setFrom(QMailAddress("0404404040"));
     msg1.setTo(QMailAddress("0404040404"));
     msg1.setSubject("Where are you?");
-    msg1.setDate(QMailTimeStamp(QDate::currentDate().startOfDay()));
-    msg1.setReceivedDate(QMailTimeStamp(QDate::currentDate().startOfDay()));
+    msg1.setDate(QMailTimeStamp(QDateTime(QDate::currentDate())));
+    msg1.setReceivedDate(QMailTimeStamp(QDateTime(QDate::currentDate())));
     msg1.setStatus(QMailMessage::Incoming, true);
     msg1.setStatus(QMailMessage::New, false);
     msg1.setStatus(QMailMessage::Read, true);
@@ -172,8 +172,8 @@ void tst_StorageManager::initTestCase()
     msg2.setTo(QMailAddress("old@example.org"));
     msg2.setCc(QMailAddressList() << QMailAddress("anotherguy@example.org"));
     msg2.setSubject("email message test");
-    msg2.setDate(QMailTimeStamp(QDate::currentDate().startOfDay()));
-    msg2.setReceivedDate(QMailTimeStamp(QDate::currentDate().startOfDay()));
+    msg2.setDate(QMailTimeStamp(QDateTime(QDate::currentDate())));
+    msg2.setReceivedDate(QMailTimeStamp(QDateTime(QDate::currentDate())));
     msg2.setStatus(QMailMessage::Incoming, true);
     msg2.setStatus(QMailMessage::New, true);
     msg2.setStatus(QMailMessage::Read, false);
@@ -211,8 +211,8 @@ void tst_StorageManager::test_add()
     msg3.setTo(QMailAddress("old@example.org"));
     msg3.setCc(QMailAddressList() << QMailAddress("anotherguy@example.org"));
     msg3.setSubject("email message test");
-    msg3.setDate(QMailTimeStamp(QDate::currentDate().startOfDay()));
-    msg3.setReceivedDate(QMailTimeStamp(QDate::currentDate().startOfDay()));
+    msg3.setDate(QMailTimeStamp(QDateTime(QDate::currentDate())));
+    msg3.setReceivedDate(QMailTimeStamp(QDateTime(QDate::currentDate())));
     msg3.setStatus(QMailMessage::Incoming, true);
     msg3.setStatus(QMailMessage::New, true);
     msg3.setStatus(QMailMessage::Read, false);
@@ -235,8 +235,8 @@ void tst_StorageManager::test_remove()
     msg4.setTo(QMailAddress("old@example.org"));
     msg4.setCc(QMailAddressList() << QMailAddress("anotherguy@example.org"));
     msg4.setSubject("email message test");
-    msg4.setDate(QMailTimeStamp(QDate::currentDate().startOfDay()));
-    msg4.setReceivedDate(QMailTimeStamp(QDate::currentDate().startOfDay()));
+    msg4.setDate(QMailTimeStamp(QDateTime(QDate::currentDate())));
+    msg4.setReceivedDate(QMailTimeStamp(QDateTime(QDate::currentDate())));
     msg4.setStatus(QMailMessage::Incoming, true);
     msg4.setStatus(QMailMessage::New, true);
     msg4.setStatus(QMailMessage::Read, false);
